scripts/core/EXT_Exp_CounterBasedEvents.rst

```diff
@@ -0,0 +1,56 @@
+<%
+import re
+from templates import helper as th
+%><%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+%>
+:orphan:
+
+.. _ZE_experimental_event_pool_counter_based:
+
+=========================
+ Counter-Based Event Pools Extension
+=========================
+
+API
+----
+ 
+* Enumerations
+
+    * ${x}_event_pool_counter_based_exp_flags_t
+    * ${x}_event_pool_counter_based_exp_version_t
+
+* Structures
+
+    * ${x}_event_pool_counter_based_exp_desc_t
+
+Counter-Based Events
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, events in Level Zero contains one of two states: signaled or not signaled.
+Signaling of an event, as well as resetting its state, can be done from either host or
+device.
+
+Another way of tracking the state of an event is by using a counter, which is incremented
+every time a task has completed in the device. Using a counter-based event may provide
+Level Zero driver implementations with the opportunity for both functional and performance
+optimizations.
+
+The following recommendations and restrictions apply to counter-based events:
+
+- Counter-based events can be used only with in-orders lists, whether those are regular or immediate.
+- Counter-based events must not be reset, i.e., a call to ${x}EventHostReset or ${x}CommandListAppendEventReset
+  is not allowed.
+- Counter-based events must not be signaled from host, i.e., a call to ${x}EventHostSignal is not allowed.
+- Counter-based events may be reused multiple times without a need for reset.
+- Counter-based events may be used on multiple command lists.
+- Querying a counter-based event queries only the last saved counter value from the last command list that incremented it,
+  i.e., a signaled counter-based event always represents the completion of the last call to which it was passed as signal event.
+- Synchronizing on a counter-based event waits only the last saved counter value from the last command list that incremented it.
+- A counter-based event may be passed as signaling event for a new append call without needing to wait for the signaling of
+  the last call where it was used.
```

Comment:
do we want to add notes on effect of "Timestamp" flags with counter events? I think timestamp flags (even if set) would be ignored right? or we want to forbid user from setting it and return error?

also- do we expect IPC events to continue to work with counter events?

Reply:
timestamps and IPC are orthogonal features, I dont think they cannot work, However, if that's the case, we can add note in the documentation in future follow ups.